### PR TITLE
Improved zint codec

### PR DESCRIPTION
**Pull request by Mallets**
Implement changes as proposed in https://github.com/eclipse-zenoh/zenoh/pull/838.

### CODE REVIEW

**Comment by Mallets** [comment]
@jean-roland would you mind review it?

File: src/protocol/codec/network.c [hunk 1]
```diff
@@ -290,10 +290,10 @@
         _Z_RETURN_IF_ERR(_z_uint8_encode(wbf, extheader));
         uint8_t zidlen = _z_id_len(msg->_ext_responder._zid);
         extheader = (zidlen - 1) << 4;
-        _Z_RETURN_IF_ERR(_z_zint_encode(wbf, zidlen + 1 + _z_zint_len(msg->_ext_responder._eid)));
+        _Z_RETURN_IF_ERR(_z_zsize_encode(wbf, zidlen + 1 + _z_zint_len(msg->_ext_responder._eid)));
```
[the remark this answers is not in the document]
Comment (jean-roland) [reply src/protocol/codec/network.c:293]: Ah yeah the dreaded (int)1 added to unsigned values, annoying but I'd rather not have this warning:

`_Z_RETURN_IF_ERR(_z_zsize_encode(wbf, zidlen + (unsigned int)1 + _z_zint_len(msg->_ext_responder._eid)));`

File: tests/z_msgcodec_test.c [hunk 2]
```diff
@@ -1615,16 +1642,20 @@ _z_network_message_t gen_net_msg(void) {
             return gen_declare_message();
         } break;
         case 1: {
+            printf("Gen Push\n");
```
Comment (jean-roland) [inline tests/z_msgcodec_test.c:1645]: Not sure we need to keep those 4 prints

File: src/protocol/codec/network.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mallets) [reply src/protocol/codec/network.c]: Done as well in other parts of the code where the same pattern was present.

**Review by jean-roland — APPROVED** [review]

**Pull request merged**